relay-event-schema/src/protocol/span.rs

```diff
@@ -718,6 +722,34 @@ impl Getter for SpanData {
     }
 }
 
+/// A link from a span to another span.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+pub struct SpanLink {
+    /// The trace id of the linked span
+    #[metastructure(required = true, trim = false)]
+    pub trace_id: Annotated<TraceId>,
+
+    /// The span id of the linked span
+    #[metastructure(required = true, trim = false)]
+    pub span_id: Annotated<SpanId>,
+
+    /// The parent span id of the linked span
+    #[metastructure(trim = false)]
+    pub parent_span_id: Annotated<SpanId>,
+
+    /// Whether the linked span was positively/negatively sampled
+    #[metastructure(trim = false)]
+    pub sampled: Annotated<bool>,
+
+    /// Span link attributes, similar to span attributes/data
+    #[metastructure(pii = "maybe", trim = false)]
+    pub attributes: Annotated<Object<Value>>,
```

Comment:
If you want to validate the `sentry.link.type` attribute, you can define `attributes` as a type `SpanLinkAttributes` that contains

```rust
struct SpanLinkAttributes {
    #[metastructure(field = "sentry.link.type", ...)]
    pub link_type: Annotated<LinkType>, // `LinkType` being an enum
    #[metastructure(additional_properties, retain = true, ...)]
    pub other: Object<Value>,
}

Reply:
as discussed offline, we'll for now not apply validation to avoid an event being dropped because of a malformed attribute. But good to know what the consequences are, thanks! :D